### DISCUSSION TITLE
[4.x] craft/web/ErrorHandler::renderException include previous Exception as Yii2 did it in Craft 3

### DISCUSSION THREAD

**Discussion by Anubarak** [craftcms/cms#11694] [Ideas] [answered]
Very simple: in Craft 3 you just called `parent::renderException` in your [ErrorHandler](https://github.com/craftcms/cms/blob/3.7.0/src/web/ErrorHandler.php#L147), now you don't do that anymore for JSON responses.
Yii2 added the previous Exception if there is one https://github.com/yiisoft/yii2/blob/master/framework/web/ErrorHandler.php#L169-L171

```PHP
if (($prev = $exception->getPrevious()) !== null) {
    $array['previous'] = $this->convertExceptionToArray($prev);
}
```

But Craft 4 not
https://github.com/craftcms/cms/blob/develop/src/web/ErrorHandler.php#L130-L142
```PHP
$response->data = [
	'message' => $exception->getMessage(),
	'exception' => get_class($exception),
	'file' => $exception->getFile(),
	'line' => $exception->getLine(),
	'trace' => array_map(function($step) {
		unset($step['args']);
		return $step;
	}, $exception->getTrace()),

	// TODO: remove in v5; error message should only be in `message`
	'error' => $exception->getMessage(),
];
```

Could you please add the previous as well like Yii2 did it before?

For the time being, I just added the following
```PHP
if(($prev = $exception->getPrevious()) !== null){
    $response->data['previous'] = [
        'message' => $prev->getMessage(),
        'exception' => \get_class($prev),
        'file' => $prev->getFile(),
        'line' => $prev->getLine(),
        'trace' => array_map(function($step) {
            unset($step['args']);
            return $step;
        }, $prev->getTrace()),
    ];
}
```

**Comment by brandonkelly** [comment, marked as answer]
Thanks for pointing that out!

This was also the case in Craft 3 – added custom JSON responses in 3.6.0 (#7406) and moved to the ErrorHandler in 3.7.17 (04bb889b0335eab80d878953ec5f976912e4a003) – so I resolved this for the next Craft 3 release as well as Craft 4. (481e7848ab1c2e405814d89714d4bac536fb205e)

**Reply by Anubarak** [reply]
Awesome thank you very much - that will really help me

**Reply by brandonkelly** [reply]
3.7.49 and 4.2.0 are out now with that change ✨
